core/src/main/scala/com/avsystem/commons/misc/Opt.scala

```diff
@@ -14,6 +14,8 @@ object Opt {
     if (value != null) new Opt[A](value)
     else throw new NullPointerException
 
+  def when[A](cond: Boolean)(value: => Opt[A]): Opt[A] = if (cond) value else Opt.Empty
+
```

Comment:
If we're adding `when`, we should probably add `unless`.

Reply:
++